src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java

```diff
@@ -211,6 +211,21 @@ public String getSuggestedFileName() {
      */
     public String getSuggestedFileName(String extension) {
         String targetFileName = FileUtil.createFileNameFromPattern(databaseContext.getDatabase(), entry, filePreferences.getFileNamePattern()).trim();
+        if ((targetFileName.isEmpty() || "-".equals(targetFileName)) && linkedFile.isOnlineLink()) {
+            String oldFileName = linkedFile.getLink();
+            int lastSlashIndex = oldFileName.lastIndexOf('/');
+            if (lastSlashIndex >= 0 && lastSlashIndex < oldFileName.length() - 1) {
```

Comment:
You might want to check some methods in FileUtil. e.g. getFileExtension or getBaseName